openbsd-compat/defines.h

```diff
@@ -418,6 +418,10 @@ typedef uint16_t	in_port_t;
 # define EWOULDBLOCK EAGAIN
 #endif
 
+#ifndef EFTYPE
+# define EFTYPE EMEDIUMTYPE
+#endif
+
```

Comment:
ok by me but I think @ericfaurot should have used another errno, we can re-assess that later, EFTYPE is only used in one particular case and your fix is acceptable.

Reply:
On OS without EFTYPE, packager should add '-DEFTYPE=ESOMETHING' on CFLAGS

Reply:
fair enough, either way I think the _real_ fix would be to pick another errno since this one is not really the best choice for that particular use-case it was used for.

@rkitover I agree with bsdsx, using a define to redefine it would be less intrusive of a hack

Reply:
The packager certainly can do this, but this will put a lot of people off who are just trying to compile the app on linux and use it and may not understand how to do this.

I think some kind of automatic configure thing would be good, but up to you.